internal/api/commitment.go

```diff
@@ -1668,3 +2122,123 @@ func (p *v1Provider) UpdateCommitmentDuration(w http.ResponseWriter, r *http.Req
+			}
+
+			if serviceInfo.Resources[resourceName].HandlesCommitments {
+				_, exists := remoteCommitmentChanges.ByProject[projectUUID]
+				if !exists {
+					remoteCommitmentChanges.ByProject[projectUUID] = liquid.ProjectCommitmentChangeset{
+						ByResource: make(map[liquid.ResourceName]liquid.ResourceCommitmentChangeset),
+					}
+				}
+				remoteCommitmentChanges.ByProject[projectUUID].ByResource[resourceName] = resourceCommitmentChangeset
+				continue
+			}
+			_, exists := localCommitmentChanges.ByProject[projectUUID]
+			if !exists {
+				localCommitmentChanges.ByProject[projectUUID] = liquid.ProjectCommitmentChangeset{
+					ByResource: make(map[liquid.ResourceName]liquid.ResourceCommitmentChangeset),
+				}
+			}
+			localCommitmentChanges.ByProject[projectUUID].ByResource[resourceName] = resourceCommitmentChangeset
+		}
+	}
+	for projectUUID, projectCommitmentChangeset := range localCommitmentChanges.ByProject {
+		if serviceInfo.CommitmentHandlingNeedsProjectMetadata {
+			pcs := projectCommitmentChangeset
+			pcs.ProjectMetadata = req.ByProject[projectUUID].ProjectMetadata
+			localCommitmentChanges.ByProject[projectUUID] = pcs
+		}
+	}
+	for projectUUID, remoteCommitmentChangeset := range remoteCommitmentChanges.ByProject {
+		if serviceInfo.CommitmentHandlingNeedsProjectMetadata {
+			rcs := remoteCommitmentChangeset
+			rcs.ProjectMetadata = req.ByProject[projectUUID].ProjectMetadata
+			remoteCommitmentChanges.ByProject[projectUUID] = rcs
+		}
+	}
+
+	// check remote
+	var liquidClient core.LiquidClient
+	c := p.Cluster
+	if len(c.LiquidConnections) == 0 {
+		// find the right ServiceType
+		liquidServiceType := c.Config.Liquids[serviceType].LiquidServiceType
+		liquidClient, err = core.NewLiquidClient(c.Provider, c.EO, liquidapi.ClientOpts{ServiceType: liquidServiceType})
+		if err != nil {
+			return result, fmt.Errorf("failed to create LiquidClient for service %s: %w", liquidServiceType, err)
+		}
+	} else {
+		liquidClient = c.LiquidConnections[serviceType].LiquidClient
+	}
+	commitmentChangeResponse, err := liquidClient.ChangeCommitments(ctx, remoteCommitmentChanges)
+	if err != nil {
+		return result, fmt.Errorf("failed to retrieve liquid ChangeCommitment response for service %s: %w", serviceType, err)
+	}
+	if commitmentChangeResponse.RejectionReason != "" {
+		return commitmentChangeResponse, nil
+	}
+
+	// check local
+	canAcceptLocally, err := datamodel.CanAcceptCommitmentChangeRequest(localCommitmentChanges, serviceType, p.Cluster, dbi)
+	if err != nil {
+		return result, fmt.Errorf("failed to check local ChangeCommitment: %w", err)
+	}
+	if !canAcceptLocally {
+		return liquid.CommitmentChangeResponse{
+			RejectionReason: "not enough capacity!",
+			RetryAt:         None[time.Time](),
+		}, nil
+	}
+
+	return result, nil
+}
+
+func LiquidProjectMetadataFromDBProject(dbProject db.Project, domain db.Domain, serviceInfo liquid.ServiceInfo) Option[liquid.ProjectMetadata] {
```

Comment:
Exported functions should have a docstring. But since these are small helpers for the current file only, they can be made private:
```suggestion
func liquidProjectMetadataFromDBProject(dbProject db.Project, domain db.Domain, serviceInfo liquid.ServiceInfo) Option[liquid.ProjectMetadata] {
```

Same for EvaluateRetryHeader below.

Reply:
I am so 👌🏻 close to building a lint for this ^^